lib_eio_linux/eio_linux.mli

```diff
@@ -248,4 +248,29 @@ module Low_level : sig
   (** [getaddrinfo host] returns a list of IP addresses for [host]. [host] is either a domain name or
       an ipaddress. *)
 
+  module Process : sig
+    type t
+    (** A subprocess *)
+
+    val spawn :
+      ?env:Spawn.Env.t ->
+      ?cwd:Spawn.Working_dir.t ->
+      ?stdin:FD.t ->
+      ?stdout:FD.t ->
+      ?stderr:FD.t ->
+      sw:Switch.t ->
+      string ->
+      string list ->
+      t
+    (** Spawns a subprocess. By default all of the optional arguments are inherited from the
+        calling process. If the process has not finished when the switch is released, the process
+        will be sent [Sys.sigkill]. *)
+
+    val await_exit : t -> Unix.process_status
+    (** [await_exit t] waits for the process [t] to exit. This blocks the fiber until the process
+        has finished. *)
+
+    val send_signal : t -> int -> unit
```

Comment:
similarly (although it's a bit weirder), this could just be `kill` as it has the same semantics as `Unix.kill`.  Easier to find in the interface if it's got a similar name.

Reply:
Hmm, I intentionally didn't go with `kill` because it seemed unnecessarily violent (see [this draft](https://datatracker.ietf.org/doc/draft-knodel-terminology/) and it's conclusions on reflecting on the type of metaphors we use) and (to someone with less experience like me) the connection to `Unix.kill` wasn't that obvious (e.g. why does `Unix.kill` need to be passed `Sys.sigkill` (my conclusion, `kill` is poorly named) ?). It also now uses `pidfd_send_signal`, but at the same time if it makes the API easier to use I'm happy to be persuaded if there's consensus that it would be better :))

Reply:
I'm entirely in agreement with the importance of avoiding violent language in systems programming.  I also dislike typing long function names. How about just `Process.signal`?

Reply:
Agreed, done in https://github.com/ocaml-multicore/eio/pull/435/commits/f6bdf04b1307cd1190f760de4fdd70049f32d7e3